src/som/compiler/MixinDefinition.java

```diff
@@ -851,12 +853,20 @@ public static boolean isInstance(final TruffleObject obj) {
     return obj instanceof MixinDefinition;
   }
 
+  /**
+   * This method provides a String that can be used to identify a MixinDefinition.
+   * The String takes a shape like this: "relativePath:module.class.nestedClass"
+   * 
+   * @return the fully qualified name of this MixinDefinition
+   */
   public String getIdentifier() {
     MixinDefinition outer = getOuterMixinDefinition();
     if (outer != null) {
       return outer.getIdentifier() + "." + this.name.getString();
     } else if (this.isModule && this.sourceSection != null) {
-      return this.sourceSection.getSource().getPath() + ":" + this.name.getString();
+      Path absolute = Paths.get(this.sourceSection.getSource().getURI());
+      Path relative = Paths.get("").toAbsolutePath().relativize(absolute);
```

Comment:
This is going to change when the working directory is changed by the program, right?

Reply:
Right, do we have the SOMns path stored somewhere?

Reply:
Not really, I think. There are various things passed in as `-D` flags or parameter in the `som` launcher script. For instance the kernel file, platform file, and tools directory. Perhaps we want the core-lib folder as relative root, too?

Reply:
We don't need a new parameter for this, we can just use the directory containing the kernel file.